### PR TITLE
Kernel/LibPthread: Plumb futex wait timeout support throughout.

### PR DESCRIPTION
**Pull request by bgianfo**
This change brings basic support for timeouts to the futex system call when invoked with the FUTEX_WAIT option. It's currently implemented by taking advantage of the timer queue to wake
the thread on timeout.

With that implemented LibPthread needed to be fixed to actually pass the option through.
A test case was added to very the feature works.

### CODE REVIEW

File: Kernel/Thread.cpp [hunk 1]
```diff
@@ -884,14 +885,21 @@ const LogStream& operator<<(const LogStream& stream, const Thread& value)
...
     if (lock)
         *lock = false;
     set_state(State::Queued);
     queue.enqueue(*current);
+
+    u64 timer_id = 0;
+    if (timeout)
+        timer_id = TimerQueue::the().add_timer(*timeout, [&]() {
+            wake_from_queue();
+        });
```
Comment (awesomekling) [inline Kernel/Thread.cpp:901]: Please wrap multi-line consequent statements in `{ }` even if the compiler allows you not to.

File: Kernel/Process.cpp [hunk 2]
```diff
@@ -4615,21 +4628,34 @@ int Process::sys$futex(const Syscall::SC_futex_params* user_params)
...
         copy_from_user(&user_value, userspace_address);
         if (user_value != value)
             return -EAGAIN;
+
+        timespec ts_abstimeout { 0, 0 };
+        if (user_timeout && !validate_read_and_copy_typed(&ts_abstimeout, user_timeout))
+            return -EFAULT;
+
+        WaitQueue& wait_queue = futex_queue(userspace_address);
+        timeval* optional_timeout = nullptr;;
```
Comment (awesomekling) [inline Kernel/Process.cpp:4644]: Double semicolon here :)

File: Kernel/Thread.cpp (hunk 1 above)
Comment (bgianfo) [reply Kernel/Thread.cpp]: Ironically, I thought I was matching your style here, fixed. :+1:

File: Kernel/Process.cpp (hunk 2 above)
Comment (bgianfo) [reply Kernel/Process.cpp]: whoops, nice catch!

**Pull request merged**